### ISSUE TITLE
doesn't impl `clone` and `FromStr`

### ISSUE THREAD

**Issue by steveeJ** [fancy-regex/fancy-regex#71]
These two trait impls would be convenient for building CLI applications that use structs for passing on arguments to subcommands.

Without these I'm using the following workarounds.

Custom parsing function with `StructOpt`. This assumes that `,` isn't used in the pattern itself, which of course is arbitrary and doesn't work in the general case.
```rust
    #[derive(StructOpt, Debug)]
    pub(crate) struct Check {
        #[structopt(long, default_value = "", parse(try_from_str = parse_regexmatch))]
        pub(crate) packages_filters: fancy_regex::Regex,
    }

    fn parse_regexmatch(input: &str) -> Fallible<fancy_regex::Regex> {
        let set = fancy_regex::Regex::new(
            &input
                .split(",")
                .filter(|s| !s.is_empty())
                .collect::<String>(),
        )?;
        Ok(set)
    }
```

Using `Regex::new` with `Regex::as_str()` as input. Being able to just call `Regex::clone` would be nice.
```rust
...
                packages_filters: fancy_regex::Regex::new(self.packages_filters.as_str())
                    .expect("regex is valid"),
...
```

**Comment by robinst** [comment]
Hey! I just released 0.6.0 which should fix your issue: https://github.com/fancy-regex/fancy-regex/blob/main/CHANGELOG.md#060---2021-05-17

Let me know if it doesn't.

**Comment by steveeJ** [comment]
Works great, thank you @robinst!
